compiler/rustc_mir/src/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -1357,6 +1357,18 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             ),
         };
         err.span_note(constraint_span, &msg);
+        if let ConstraintCategory::CallArgument = category {
+            if let Some(generator_kind) = use_span.generator_kind() {
+                if let GeneratorKind::Async(_) = generator_kind {
```

Comment:
We can get rid of some unnecessary nesting by writing

```suggestion
        if let (ConstraintCategory::CallArgument, Some(GeneratorKind::Async(_))) = (category, use_span.generator_kind()) {
```